src/Server/Tasks/Numeral.fs

```diff
@@ -5,12 +5,30 @@ open FSharp.Control.Tasks.V2
 open Giraffe
 open Microsoft.AspNetCore.Http
 
-open Core.Numerals
 open Server.Tasks.Utils
- 
+open Core.Numerals
+open Common.Utils
+open Common.Numerals
+
+let getMinMaxRange = function
+    | From0         -> (0, Int32.MaxValue)
+    | From0To100    -> (0, 100)
+    | From100To1000 -> (100, 1000)
+    | From1000To1000000 -> 
+        (1000, 1000000)
+    | From1000000 -> (1000000, Int32.MaxValue)
+
+let getRandomNumeral (min, max) =
+    Random().Next(min, max)
+
 let getNumeralCardinalsTask next (ctx: HttpContext) =
     task {
-        let number = Random().Next()
+        let range =
+            ctx.TryGetQueryStringValue "range"
+            |> Option.map parseUnionCase<Range>
+            |> Option.defaultValue Range.From0
+        let minMaxRange = getMinMaxRange range
```

Comment:
Naming: there is no clear difference between `range` and `minMaxRange`. Consider renaming `getMinMaxRange` to `getRange` and pipe it further:
```fsharp
let range =
    ctx.TryGetQueryStringValue "range"
    |> Option.map parseUnionCase<Range>
    |> Option.defaultValue Range.All
    |> getRange
```